src/muya/lib/parser/marked/lexer.js

```diff
@@ -255,21 +255,41 @@ Lexer.prototype.token = function (src, top) {
         })
 
         // Changing the bullet or ordered list delimiter starts a new list (CommonMark 264 and 265)
+        // Changing to/from task list item from/to bullet, starts a new lit(work for marktext)
         //   - unordered, unordered --> bull !== newBull --> new list (e.g "-" --> "*")
         //   - ordered, ordered --> lastChar !== lastChar --> new list (e.g "." --> ")")
         //   - else --> new list (e.g. ordered --> unordered)
         const newIsOrdered = bull.length > 1 && /\d{1,9}/.test(newBull)
-        if (i !== 0 &&
-          ((!isOrdered && !newIsOrdered && bull !== newBull) ||
-          (isOrdered && newIsOrdered && bull.slice(-1) !== newBull.slice(-1)) ||
-          ((isOrdered && !newIsOrdered) || (!isOrdered && newIsOrdered)))) {
+
+        if (!newIsOrdered && this.options.gfm) {
+          checked = this.rules.checkbox.exec(item)
+          if (checked) {
+            checked = checked[1] === 'x' || checked[1] === 'X'
+            item = item.replace(this.rules.checkbox, '')
+            newIsTaskListItem = true
+          } else {
+            checked = undefined
+          }
+        }
+        if (i === 0) {
+          isTaskListItem = newIsTaskListItem
+        } else if (
```

Comment:
Please move the comment block `Changing the bullet or ordered list delimiter starts a new list (CommonMark 264 and 265)...` from above here.